feature #869: create endpoint for user creation

ArtemiiLiakh opened the pull request:
- created createUser endpoint
- created UserByAdminDTO for the endpoint
- added admin PERMISSION

closes fictadvisor/fictadvisor#65

PavloVlasiuk commented on src/v2/api/controllers/UserController.ts:
```diff
@@ -68,6 +69,44 @@ export class UserController {
+  @Post('/createUser')
+  async createUser (@Body() body: UserByAdminDTO) {
+    return this.userMapper.getUser(
+      await this.userService.createUserByAdmin(body)
```
create an intermediate variable for data that are returned by createUserByAdmin method and then pass it to getUser method

PavloVlasiuk commented on src/v2/api/controllers/UserController.ts:
```diff
@@ -68,6 +69,44 @@ export class UserController {
 
+  @ApiBearerAuth()
+  @ApiOkResponse({
+    type: UserResponse,
```
make avatar an optional field in UserResponse class

ArtemiiLiakh committed:
feature #869: create endpoint for user creation

- created createUser endpoint
- created UserByAdminDTO for the endpoint
- added admin PERMISSION

closes #869

fokaaas commented:
closed as outdated
